openedx/core/lib/xblock_utils.py

```diff
@@ -211,6 +212,12 @@ def grade_histogram(module_id):
     return grades
 
 
+def sanitize_html_id(html_id):
+    """Template uses element_id in js function names, so can't allow dashes and colons"""
```

Comment:
Comment should be on its own line.  I need to do some more research, but I'm not clear on where the html_id is getting defined.  It seems like there are many more illegal characters for js function names, if that is the root of the problem.  Is there a way to avoid the problem altogether?  Otherwise, we'll need a more complicated sanitize method:
- https://mothereff.in/js-variables
- http://stackoverflow.com/questions/1661197/what-characters-are-valid-for-javascript-variable-names
Functionas and variables might have different rules, so this may not be perfect, but you get the point hopefully.